lib/flowbber/plugins/aggregators/lcov.py

```diff
@@ -0,0 +1,266 @@
+                    "rc_overrides": ["lcov_branch_coverage=1"],
+                    "remove": ["*hello2*"]
+                }
+            }
+        ]
+    }
+
+keys
+----
+
+List of lcov sources.
+
+- **Default**: ``N/A``
+- **Optional**: ``False``
+- **Schema**:
+
+  .. code-block:: python3
+
+        schema={
+            'type': 'list',
+            'schema': {
+                'type': 'string',
+                'empty': False,
+            },
+        },
+
+- **Secret**: ``False``
+
+rc_overrides
+------------
+
+Override lcov configuration file settings.
+
+Elements should have the form ``SETTING=VALUE``.
+
+- **Default**: ``[]``
+- **Optional**: ``False``
+- **Schema**:
+
+  .. code-block:: python3
+
+     {
+         'type': 'list',
+         'schema': {
+             'type': 'string',
+             'empty': False
+         },
+     }
+
+- **Secret**: ``False``
+
+remove
+------
+
+List of patterns of files to remove from coverage computation.
+
+Patterns will be interpreted as shell wild‐card patterns.
+
+- **Default**: ``[]``
+- **Optional**: ``True``
+- **Schema**:
+
+  .. code-block:: python3
+
+     {
+         'type': 'list',
+         'schema': {
+             'type': 'string',
+             'empty': False,
+         },
+     }
+
+- **Secret**: ``False``
+
+"""
+
+from shutil import which
+from tempfile import NamedTemporaryFile
+from pathlib import Path
+
+from flowbber.components import Aggregator
+from flowbber.utils.command import run
+from flowbber.plugins.sources.lcov import LcovSource
+
+
+class LcovAggregator(Aggregator):
+
+    def declare_config(self, config):
+        config.add_option(
+            'keys',
+            optional=False,
+            schema={
+                'type': 'list',
+                'schema': {
+                    'type': 'string',
+                    'empty': False,
+                },
+            },
+        )
+
+        config.add_option(
+            'rc_overrides',
+            default=[],
+            optional=True,
+            schema={
+                'type': 'list',
+                'schema': {
+                    'type': 'string',
+                    'empty': False,
+                },
+            },
+        )
+
+        config.add_option(
+            'remove',
+            default=[],
+            optional=True,
+            schema={
+                'type': 'list',
+                'schema': {
+                    'type': 'string',
+                    'empty': False,
+                },
+            },
+        )
+
+    def accumulate(self, data):
+        # Read all sources tracefiles paths
+        tracefiles = (
+            ''.join(
+                '--add-tracefile {} '.format(data[source]['tracefile'])
+                for source in sorted(set(self.config.keys.value))
+                if source in data
+            )
+        )
+
+        if len(tracefiles) == 0:
+            raise ValueError('No valid tracefiles where found in data')
```

Comment:
tracefiles can be empty even if I define a schema, because user can specify something like keys=['something'] and data['something'] may not exists.